bionic/code_hasher.py

```diff
@@ -162,11 +174,13 @@ def _update_hash(self, hash_accumulator, obj, code_context):
                 obj_bytes=str(obj).encode(),
             )
 
-        elif isinstance(obj, (list, set, tuple)):
+        elif isinstance(obj, (list, set, tuple, frozenset)):
             if isinstance(obj, list):
                 type_prefix = TypePrefix.LIST
             elif isinstance(obj, set):
                 type_prefix = TypePrefix.SET
+            elif isinstance(obj, frozenset):
+                type_prefix = TypePrefix.FROZENSET
```

Comment:
Hmm... does the fact that frozensets (and sets) have non-deterministic order present any problems here?

Reply:
Ah yeah, we wanted to do it for both this and protocols but I think we forgot about this part. I have an idea on how to fix it. I can create an individual hash for each element and sort the hashes before adding it to the `hash_accumulator`. That way, the order would be guaranteed.

I'll do it in a follow up task to not block this TODO.